src/main/java/org/dependencytrack/tasks/repositories/AbstractMetaAnalyzer.java

```diff
@@ -33,25 +35,32 @@
 import org.dependencytrack.notification.NotificationScope;
 import org.dependencytrack.util.HttpUtil;
 
-import java.io.IOException;
-import java.net.URISyntaxException;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
+import alpine.common.logging.Logger;
+import alpine.notification.Notification;
+import alpine.notification.NotificationLevel;
 
 /**
- * Base abstract class that all IMetaAnalyzer implementations should likely extend.
+ * Base abstract class that all IMetaAnalyzer implementations should likely
+ * extend.
  *
  * @author Steve Springett
  * @since 3.1.0
  */
 public abstract class AbstractMetaAnalyzer implements IMetaAnalyzer {
 
+    protected String repositoryId;
     protected String baseUrl;
-
     protected String username;
-
     protected String password;
 
+    /**
+     * Optionally set repositoryId that can be used in a cache key for example.
+     * @param repositoryId
+     */
+    public void setRepositoryID(String repositoryId) {
```

Comment:
This should be called from `RepositoryMetaAnalyzerTask#analyze` when the analyzer is constructed. At the moment the setter is only called in tests.

Reply:
Good catch, added now.